### PR TITLE
Mark deprecation to `ubygems.rb` for RubyGems 4

### PR DESCRIPTION
**Pull request by hsbt**
# Description:

I'm going to mark deprecate status to obsoleted methods for RubyGems 4
______________

# Tasks:

- [ ] Describe the problem / feature
- [ ] Write tests
- [ ] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by hsbt** [comment]
@segiddins I don't know. `ubygems.rb` will never use on the current supported versions of Ruby. It should be removed from this repository. 

PS. I already removed its file from ruby core at Ruby 2.5.

**Comment by segiddins** [comment]
But it’s installed when I `Gem update —system`, and I don’t think we should remove it without discussing the reason for doing so

**Comment by segiddins** [comment]
> I already decided to remove it in RubyGems 4

I don’t think that’s fair — it’s for the RubyGems team to collectively make decisions, not just you or me.

As to my aversion to removing this file? I run `ruby -rubygems` frequently

**Comment by hsbt** [comment]
> I run ruby -rubygems frequently

with Ruby 1.8? I wonder why you add `-rubygems` in 2018.

**Comment by hsbt** [comment]
1. We already adopted to remove to support old rubies: https://github.com/rubygems/rubygems/issues/2191
2. I did NOT remove its file immediately. At first, We warn deprecated message at RubyGems 3. I will remove it at RubyGems 4
3. You did accept other removable pull requests for RubyGems 3 & 4.

I think your opinion is only your experience. It's not reasonable.

**Comment by segiddins** [comment]
It’s not reasonable to want a discussion, rather than just making changes, saying “a decision has been made”?

**Comment by hsbt** [comment]
@segiddins I'm waiting "your reasonable reason" yet.

After Ruby 1.9, rubygems is automatically activated by the prelude of Ruby. You should not add `-rubygems` on the technical side. I do not understand your reasons for this problem

**Comment by segiddins** [comment]
My problem isn't the change per-se, it's the assertion of _this has been decided_ when no discussion has taken place.

> You should not add -rubygems on the technical side

a) I run 1.8 occasionally
b) `-disable-gems`

**Comment by hsbt** [comment]
> a) I run 1.8 occasionally

The current master branch was not supported Ruby 1.8. It was not related this.

> b) -disable-gems

Why you add `-rubygems` with `--disable-gems`? It's not reasonable.

**Comment by segiddins** [comment]
> Why you add -rubygems with --disable-gems?

Aliases

> It was not related this.

It is related. I have scripts that are meant to work on all versions of ruby, and this feels like a breaking change for no good reason

**Comment by hsbt** [comment]
> It is related. I have scripts that are meant to work on all versions of ruby,

It's your personal problem.

**Comment by duckinator** [comment]
@segiddins is there any reasons your scripts can't be updated to use `-rrubygems` instead of `-rubygems`?

@hsbt _I'm in favor of deprecating this_, but i'd appreciate it if you would consider re-adding it to Ruby _until RubyGems removes it._ This was a feature for at least 12 years(!), and removing it abruptly like you did in Ruby 2.5 seems risky. We want to provide time for people to adjust their scripts.

File: lib/ubygems.rb [hunk 1]
```diff
@@ -7,5 +7,8 @@
 # See LICENSE.txt for permissions.
 #++
 
+# TODO: Remove this file at RubyGems 4.0
+# Based by https://bugs.ruby-lang.org/issues/14322#note-4
+warn "`ubygems.rb' is deprecated, It will be removed on or after 2018-12-01. remove `-rubygems' from your command-line"
```
Comment (duckinator) [inline lib/ubygems.rb]: @hsbt what do you think of this message?:

```
warn "`ubygems.rb' is deprecated, and will be removed on or after 2018-12-01. Remove `-rubygems' from your command-line, or use `-r rubygems' instead"
```

**Comment by duckinator** [comment]
Left a comment with some input on the warning message.

File: lib/ubygems.rb (hunk 1 above)
Comment (hsbt) [reply lib/ubygems.rb]: 👍

**Comment by shevegen** [comment]
Early deprecation notices in advance are good, IMO.

It gives people more time to transition (with a transition period
that should take some time too).

**Comment by duckinator** [comment]
@hsbt are you going to change the warning? you've got a +1 from me on the deprecation either way, but i don't want to merge it before it's ready if you're changing it.

**Comment by hsbt** [comment]
@bundlerbot r+

**Pull request merged**